### DISCUSSION TITLE
Vaultwarden has missing domains in Domain Rules

### DISCUSSION THREAD

**Discussion by Supernova3339** [dani-garcia/vaultwarden#2876] [Q&A]
Hi, I noticed vaultwarden is missing a domain for paypal.
Is this normal? There is paypal.com, paypal-search.com, but [paypal-communications.com](https:///paypal-communications.com) doesn't exist.

**Comment by jjlin** [comment]
The Vaultwarden global equivalent domains are taken from upstream Bitwarden's list at

https://github.com/bitwarden/server/blob/master/src/Core/Utilities/StaticStore.cs

Note that a lot of these entries are actually out of date and/or obsolete. I believe Bitwarden initially just took them from LastPass, but they don't really review them regularly. For example, paypal-search.com just redirects to paypal.com, so there probably shouldn't even be a PayPal entry anymore.

I don't know what your extra domain is used for (it redirects to paypal-brandcentral.com and asks for a "Brand Central" login, which doesn't sound like a normal paypal.com login), but if you think it should be there, you can open an issue or PR in their repo and make your case.

**Reply by Supernova3339** [reply]
This is where most of paypal business backend apis and the main mailserver go through.
Should I make the PR? Thanks for replying.

**Reply by BlackDex** [reply]
If you think it needs to be there, then you can always open a PR at Bitwarden to have it added. Ones they have added it we will update it from there repo.

Otherwise, login into your vault, then go to Settings > Domain Rules and make the adjustment.

**Reply by jjlin** [reply]
@Supernova3339 It doesn't sound like this is a useful entry in the equivalent domains list. The point of these lists is not to track every domain that is associated with an organization, but only the ones that

1. Users actually log in through
2. Share a common username/password

For example, Amazon has many country-specific sites that all accept the same username/password, so the equivalent domains list allows you to have a single entry for Amazon where you add a URI for `amazon.com` (or whatever your home region is), but if you went to `amazon.co.uk` or `amazon.co.jp`, autofill could use that same Amazon entry even though you didn't explicitly add those extra URIs to your entry.

For this reason, the current PayPal list is actually obsolete, since `paypal-search.com` just redirects to `paypal.com`, so there would never be a need to autofill at `paypal-search.com`. Presumably this was not true at some point in the past.
